src/main/java/com/amalitech/gpuconfigurator/service/JwtService.java

```diff
@@ -0,0 +1,137 @@
+     * with the {@code Claims::getSubject} function to get the email as the subject claim.
+     * @param jwtToken the JWT token to extract the email from
+     * @return the email address of the user, or null if the token is invalid or does not contain the subject claim
+     */
+    public String extractEmail(String jwtToken) {
+        return extractSingleClaim(jwtToken,Claims::getSubject);
+    }
+
+    /**
+     * Extracts a single claim from a JWT token using a given function.
+     * @param <T> the type of the claim to be extracted
+     * @param jwtToken the JWT token as a string
+     * @param claimsResolver the function that takes a Claims object and returns a claim of type T
+     * @return the extracted claim of type T
+     * @throws IllegalArgumentException if the jwtToken is invalid
+     */
+    public <T> T extractSingleClaim(String jwtToken, Function<Claims,T> claimsResolver){
+        try{
+            final Claims claims = extractAllClaims(jwtToken);
+            return claimsResolver.apply(claims);
+        }catch (Exception e){
+            throw new IllegalArgumentException("Invalid Jwt",e);
+        }
+    }
+
+    /**
+     * Parses a JWT token and returns its claims.
+     * @param jwtToken the JWT token as a string
+     * @return the claims of the JWT token
+     * @throws IllegalArgumentException if the jwtToken is invalid
+     */
+    private Claims extractAllClaims(String jwtToken){
+      try{
+          return Jwts
+                  .parserBuilder()
+                  .setSigningKey(getSignInKey())
+                  .build()
+                  .parseClaimsJws(jwtToken)
+                  .getBody();
+      }catch (Exception e){
+          throw new IllegalArgumentException("Invalid Jwt",e);
+      }
+
+    }
+
+    /**
+     * Generates a HMAC-SHA key for signing in using a secret.
+     *
+     * @param secret the secret used to generate the key, encoded in Base64
+     * @return a {@code Key} object representing the HMAC-SHA key
+     */
+    private Key getSignInKey() {
+        byte[] byteKey = Decoders.BASE64.decode(secret);
+        return Keys.hmacShaKeyFor(byteKey);
+    }
+
+
+    /**
+     * Generates a JSON Web Token (JWT) for authentication using extra claims and user details.
+     *
+     * @param extraClaims a map of additional claims to be included in the token
+     * @param userDetails an object containing the user's username and other information
+     * @return a string representation of the JWT
+     */
+    public String generateToken(Map<String,Object> extraClaims, UserDetails userDetails){
+        return Jwts
+                .builder()
+                .setClaims(extraClaims)
+                .setSubject(userDetails.getUsername()) //getUsername() is a getter for the email in user model
+                .setIssuedAt(new Date(System.currentTimeMillis()))
+                .setExpiration(new Date(System.currentTimeMillis() + 1000 * 60 * 24))
```

Comment:
It'll be nice to have the expiration time in an env variable, so we don't need to dig into the code anytime we want to say increase or decrease the expiration of our JWT tokens